File: packages/java/src/main/java/com/ailabtools/sdk/api/BackgroundApi.java
package com.ailabtools.sdk.api;

import com.ailabtools.sdk.generated.params.CutoutUniversalBackgroundRemovalParams;
import com.ailabtools.sdk.generated.response.CutoutUniversalBackgroundRemovalResponse;

/** Convenience methods for common background operations. */
public final class BackgroundApi {
    private final CutoutApi cutout;

    public BackgroundApi(CutoutApi cutout) {
        this.cutout = cutout;
    }

    public CutoutUniversalBackgroundRemovalResponse remove(
            CutoutUniversalBackgroundRemovalParams params) {
        return cutout.cutoutUniversalBackgroundRemoval(params);
    }
}

File: packages/kotlin/src/main/kotlin/com/ailabtools/sdk/api/BackgroundApi.kt
package com.ailabtools.sdk.api

import com.ailabtools.sdk.generated.params.CutoutUniversalBackgroundRemovalParams
import com.ailabtools.sdk.generated.response.CutoutUniversalBackgroundRemovalResponse

/** Convenience operations for common background-removal workflows. */
public class BackgroundApi internal constructor(
    private val cutout: CutoutApi,
) {
    /** Removes the background from an image. */
    public suspend fun remove(
        parameters: CutoutUniversalBackgroundRemovalParams,
    ): CutoutUniversalBackgroundRemovalResponse =
        cutout.cutoutUniversalBackgroundRemoval(parameters)
}
